src/window/window_wrapper.rs

```diff
@@ -211,6 +232,11 @@ impl WinitWindowWrapper {
         self.windowed_context.window().set_title(&self.title);
     }
 
+    pub fn handle_theme_changed(&mut self, new_theme: Option<Theme>) {
+        self.theme = new_theme;
+        self.windowed_context.window().set_theme(self.theme);
```

Comment:
I think this might call an infinite recursion, at least on some platforms, since `WindowEvent` is windows specific, and should be fired when the theme of a window is changed.

And by quickly looking at the code it looks like the macOS implementation checks the theme from the window https://github.com/rust-windowing/winit/blob/4f6fd44c6ce9d5020360257140aee351813664b4/src/platform_impl/macos/window_delegate.rs#L400-L404

So, I think this should be removed for now, and that winit handles the theme change by itself already internally.

At some point, we might want to notify Neovim about this change though, and let it automatically switch `background`, unless it's set to something in the user configuration.

Reply:
I just realized that I'm wrong, this is our own theme changed event, and not the winit one. I think we need to rename `WindowCommand` to something else at some point, it's too similar to `WindowEvent`

Reply:
> I just realized that I'm wrong, this is our own theme changed event, and not the winit one. I think we need to rename `WindowCommand` to something else at some point, it's too similar to `WindowEvent`

Yeah, I was just about to ask you what the dead loop is. I put it at `WindowCommand` because, I read the code carefully and I find `WindowCommand::TitleChanged` there. Maybe you can first follow `WindowCommand::TitleChanged` to get the logic clearly.

As I can understand, the `WindowCommand` is a way to set some window properties from other components, like window title is asked to change, since not every components hold a ref to window object.

Reply:
I think the name `WindowCommand` comes from the fact that those events target our `WinitWindowWrapper`. And is usually something that needs to be further sent to winit. While our draw commands are more about the Neovide grid. 

So you did correctly here, and I was not not reading the code carefully enough before writing the message.